rs/ic-canisters/src/lib.rs

```diff
@@ -83,13 +84,17 @@ impl IcAgentCanisterClient {
     }
 
     fn build_agent(url: Url, identity: Box<dyn Identity>) -> anyhow::Result<Self> {
-        Ok(Self {
-            agent: Agent::builder()
-                .with_identity(identity)
-                .with_transport(ReqwestTransport::create(url)?)
-                .with_verify_query_signatures(false)
-                .build()?,
-        })
+        let client = reqwest::Client::builder()
+            .use_rustls_tls()
+            .timeout(Duration::from_secs(30))
+            .build()
+            .expect("Could not create HTTP client.");
+        let agent = Agent::builder()
+            .with_identity(identity)
+            .with_transport(ReqwestTransport::create_with_client(url, client)?)
+            .with_verify_query_signatures(false)
+            .build()?;
+        Ok(Self { agent })
```

Comment:
Looks like we reuse this functionality a lot. Can we call it in other parts of code? Looks like it is basically the same code

Reply:
Maybe we could create our own wrapper for this, but it would actually be better to have https://github.com/dfinity/agent-rs/pull/277 merged in agent-rs. Then we wouldn't need this boilerplate code. So I'll try to push for that instead.